### PR TITLE
HDDS-4126. Freon key generator should support >2GB files.

### PR DESCRIPTION
**Pull request by ArafatCloudera**
## What changes were proposed in this pull request?

fileSize parameter in HadoopFsGenerator is an int which limits it to max 2GB. But works well with long.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4126

## How was this patch tested?

Na

### CODE REVIEW

**Review by jojochuang — APPROVED** [review]
+1 LGTM
This is trivial so no test is required.

**Review by adoroszlai — APPROVED** [review]
Thanks @ArafatCloudera for the improvement.

Tested:

```
$ ozone sh volume create vol1
$ ozone sh bucket create --layout FILE_SYSTEM_OPTIMIZED vol1/fso
$ ozone freon dfsg -n1 -t1 --path ofs://om/vol1/fso -p test -s 4294967296
...
$ ozone fs -ls ofs://om/vol1/fso/test
Found 1 items
-rw-rw-rw-   3 hadoop hadoop 4294967296 2022-02-08 13:34 ofs://om/vol1/fso/test/0
```

**Review by adoroszlai — COMMENTED** [review]
@ArafatCloudera I have just noticed that there are a few other cases where the same limit applies:

```
hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/StreamingGenerator.java
66:  private int fileSize;

hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/S3KeyGenerator.java
73:  private int fileSize;

hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopDirTreeGenerator.java
73:  private int fileSizeInBytes;

hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/containergenerator/BaseGenerator.java
44:  private int keySize;
```

Would you mind changing these as well to `long`?

**Comment by ArafatCloudera** [comment]
> @ArafatCloudera I have just noticed that there are a few other cases where the same limit applies:
> 
> ```
> hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/StreamingGenerator.java
> 66:  private int fileSize;
> 
> hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/S3KeyGenerator.java
> 73:  private int fileSize;
> 
> hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopDirTreeGenerator.java
> 73:  private int fileSizeInBytes;
> 
> hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/containergenerator/BaseGenerator.java
> 44:  private int keySize;
> ```
> 
> Would you mind changing these as well to `long`?

Sure will get it done !!

**Comment by ArafatCloudera** [comment]
@adoroszlai I could only make changes for StreamingGenerator.java & HadoopDirTreeGenerator.java 

I could not make changes to S3KeyGenerator.java  because the **RandomStringUtils.randomAscii(fileSize)** function on **line:122** takes only Integer as an argument.
```
hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/S3KeyGenerator.java
73:  private int fileSize;
```

I could not make changes to BaseGenerator.java because a lot of the code would have to be rewritten in other files.
```
hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/containergenerator/BaseGenerator.java
44:  private int keySize;
```

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/containergenerator/GeneratorDatanode.java [hunk 1]
```diff
@@ -256,7 +256,7 @@ public void generateData(long index) throws Exception {
         //loop to create multiple chunks per blocks
         while (writtenBytes < getKeySize()) {
           int currentChunkSize =
-              Math.min(getKeySize() - writtenBytes, chunkSize);
+                  Math.min(getKeySize() - writtenBytes, chunkSize);
```
Comment (adoroszlai) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/containergenerator/GeneratorDatanode.java:259]: `GeneratorDatanode` only has whitespace change.  Can you please omit it from the patch?

**Comment by adoroszlai** [comment]
> I could only make changes for StreamingGenerator.java & HadoopDirTreeGenerator.java

That's fine.  Thanks for checking.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/containergenerator/GeneratorDatanode.java (hunk 1 above)
Comment (ArafatCloudera) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/containergenerator/GeneratorDatanode.java]: Yea sorry about that, I have removed it

**Comment by adoroszlai** [comment]
Thanks @ArafatCloudera for the patch, @jojochuang for the review.

**Pull request merged**